simple-dns/src/dns/rdata/svcb.rs

```diff
@@ -0,0 +1,326 @@
+                b"\x00\x00\x03foo\x07example\x03com\x00",
+                SVCB {
+                    priority: 0,
+                    target: Name::new_unchecked("foo.example.com"),
+                    params: vec![],
+                },
+            ),
+            (
+                "D.2.3. TargetName Is '.'",
+                b"\x00\x01\x00",
+                SVCB {
+                    priority: 1,
+                    target: Name::new_unchecked(""),
+                    params: vec![],
+                },
+            ),
+            (
+                "D.2.4. Specified a Port",
+                b"\x00\x10\x03foo\x07example\x03com\x00\x00\x03\x00\x02\x00\x35",
+                SVCB {
+                    priority: 16,
+                    target: Name::new_unchecked("foo.example.com"),
+                    params: vec![
+                        SvcParam::port(53),
+                    ],
+                },
+            ),
+            (
+                "D.2.6. A Generic Key and Quoted Value with a Decimal Escape",
+                b"\x00\x01\x03foo\x07example\x03com\x00\x02\x9b\x00\x09hello\xd2qoo",
+                SVCB {
+                    priority: 1,
+                    target: Name::new_unchecked("foo.example.com"),
+                    params: vec![
+                        SvcParam::new(667, b"hello\xd2qoo").unwrap(),
+                    ],
+                },
+            ),
+            (
+                "D.2.7. Two Quoted IPv6 Hints",
+                b"\x00\x01\x03foo\x07example\x03com\x00\x00\x06\x00\x20\
+                    \x20\x01\x0d\xb8\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x01\
+                    \x20\x01\x0d\xb8\x00\x00\x00\x00\x00\x00\x00\x00\x00\x53\x00\x01",
+                SVCB {
+                    priority: 1,
+                    target: Name::new_unchecked("foo.example.com"),
+                    params: vec![
+                        SvcParam::ipv6hint([
+                            0x2001_0db8_0000_0000_0000_0000_0000_0001,
+                            0x2001_0db8_0000_0000_0000_0000_0053_0001,
+                        ]).unwrap(),
+                    ],
+                },
+            ),
+            (
+                "D.2.10. SvcParamKey Ordering Is Arbitrary in Presentation Format but Sorted in Wire Format",
+                b"\x00\x10\x03foo\x07example\x03org\x00\
+                    \x00\x00\x00\x04\x00\x01\x00\x04\
+                    \x00\x01\x00\x09\x02h2\x05h3-19\
+                    \x00\x04\x00\x04\xc0\x00\x02\x01",
+                SVCB {
+                    priority: 16,
+                    target: Name::new_unchecked("foo.example.org"),
+                    params: vec![
+                        SvcParam::mandatory([SvcParam::ALPN, SvcParam::IPV4HINT]).unwrap(),
+                        SvcParam::alpn(["h2".try_into().unwrap(), "h3-19".try_into().unwrap()]).unwrap(),
+                        SvcParam::ipv4hint([0xc0_00_02_01]).unwrap(),
+                    ],
```

Comment:
This test should have the key order inverted to test the sorting/validation of the param keys